tests/admin_changelist/tests.py

```diff
@@ -1613,6 +1614,21 @@ def test_search_bar_total_link_preserves_options(self):
                     response, f'0 results (<a href="{href}">1 total</a>)'
                 )
 
+    def test_list_display_foreign_field(self):
+        parent = Parent.objects.create(name="I am your father")
+        child = Child.objects.create(name="I am your child", parent=parent)
+        GrandChild.objects.create(name="I am your grandchild", parent=child)
+        request = self.factory.get("/grandchild/")
+        request.user = self.superuser
+
+        class GrandChildAdminCustom(admin.ModelAdmin):
+            list_display = ["parent__name", "parent__parent__name"]
+
+        m = GrandChildAdminCustom(GrandChild, custom_site)
+        response = m.changelist_view(request)
+        self.assertContains(response, parent.name)
+        self.assertContains(response, child.name)
+
```

Comment:
Could you please add a test for the error case, where the requested attribute name does not exist?